plugins/mac-vth264/encoder.c

```diff
@@ -145,14 +147,26 @@ static OSStatus session_set_prop(VTCompressionSessionRef session,
 }
 
 static OSStatus session_set_bitrate(VTCompressionSessionRef session,
+				    const char *rate_control, int quality,
 				    int new_bitrate, bool limit_bitrate,
 				    int max_bitrate, float max_bitrate_window)
 {
 	OSStatus code;
 
-	SESSION_CHECK(session_set_prop_int(
-		session, kVTCompressionPropertyKey_AverageBitRate,
-		new_bitrate * 1000));
+	bool rc_constant_quality = astrcmpi(rate_control, "Constant quality") ==
+				   0;
+
+	if (rc_constant_quality) {
+		float quality_num = quality >= 100 ? 1.0 : (float)quality / 100;
```

Comment:
Why is quality not a float to begin with? The conversion of the value should happen in UI (say a user selects "100", it should be stored as "1.0" then).